File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        vector<int> last(256,-1);
        int l=0,ans=0,n=s.size();
        for(int r=0;r<n;r++){
            int c=last[s[r]];
            l=max(l,c+1);
            last[s[r]]=r;
            ans=max(ans,r-l+
            1);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> s=new HashSet<>(Arrays.stream(nums).boxed().toList());
        int ans=0;
        for(int x:nums){
if(!s.contains(x-1)){
int len=1;
while(s.contains(x+len)) len++;
ans=Math.max(ans,len);
}
        }
        return ans;
    }
}
